### DISCUSSION TITLE
GPLv2 vs of MIT license

### DISCUSSION THREAD

**Discussion by coderberry** [easepick/easepick#53] [General]
We are looking at possibly using this library in our company, however, I noticed the license is GPLv2. Unfortunately, this makes it so we can't use this incredible library. I was wondering if one of the reasons behind the GPL license is to eventually monetize this library?

Thanks!

**Comment by DennisSuitters** [comment]
Being GPL doesn't necessarily mean they are going to monetize, but from experience with other projects, people are more likely to contribute more freely with a less restrictive license such as MIT.

**Comment by wakirin** [comment]
I think GPLv2 is more fair to the open source community and the author of the project.
I'll try to explain why.
For example @coderberry said:
> We are looking at possibly using this library in our company, however, I noticed the license is GPLv2.
> Unfortunately, this makes it so we can't use this incredible library.

I'm guessing it's because they don't want to share their project with the open source community.
It's okay, I don't blame them.
It's just that if you have some kind of commercial project and you don't want to share your project with the open source community, you should buy a commercial license from the author of the project.

Thus we have three ways:
- the open source community will have more open source projects
- or authors of open-source projects can have some profit
- or use other projects under MIT or a similar license

**Reply by coderberry** [reply]
Thank you for your reply, @wakirin, and thank you for releasing this library as open source.

The problem for us with the GPL software license is its reciprocity. Any software that's derived from software licensed under the GPL and released for public consumption must also be GPL-licensed, and it must have the source code readily available. As our company is private with private source code, we cannot use this library.

**Comment by jomres** [comment]
I'm with coderberry, I'd much rather see this released as MIT. GPL is too infectious.

**Comment by kevinpapst** [comment]
+1 for MIT - I am maintaining an open source project under MIT and adding easepick would cause license issues, so I won't use it and will stick with Litepicker (which is MIT).

> or authors of open-source projects can have some profit

There are many successful MIT-licensed projects out there which generated income. GPL 2 was written at a time when open source was not driving the entire IT economy. Big frameworks are released under MIT for good reasons. GPL is mostly about sharing source code - and this can be done in many ways that will not help the open source community at all. 

Anyway: I am really glad that you share your fantastic projects, thank you!

**Comment by hajo-travanto** [comment]
Is it required to also share the code if this library is just used "as-is" in another software project without modifications or extensions being made? That would be a no-go for me, too.

Is there a list of prices somewhere for this project as an alternative? No need for special extras, but i want to avoid any legal issues and didn't see its GPL at first.

**Reply by daniel-rikowski** [reply]
> Is it required to also share the code if this library is just used "as-is" in another software project without modifications or extensions being made? That would be a no-go for me, too.

There are various aspects and opinions on that topic:

When you use a bundler and serve a minified version of the library you most definitily would have created a derived work published to a user in non-source form. That whole derived work then must be licensed under GPL and your original project (the part without the library) must be licensed at least GPL-compatible. (MIT, MPL, ...)

When using a public CDN then it becomes a question of doing so is considered "aggregation" or "linking". If it is linking then you'd have a derived work and the copyleft mechanism applies, only with aggregation you'd be (relatively) safe. The problem: Nobody seems to have a definitive answer to that distinction, as there are good points for both perspectives. Sometimes library authors clarify their stance on that particular question, but still, a lawyer might disagree...

All in all Javascript libraries with GPL are not a good fit (IMHO), because the GPL relies on C mechanisms/concepts which do not readily apply to Javascript.

**Comment by hajo-travanto** [comment]
What happens if i e.g. use a <easepick> webcomponent derived from this project and integrate that one in my software? The only "added" content would be the boilerplate code of the webcomponent toolkit that was used and i load that HTML tag into my project. So there is no real new functionality, it is just this additional "wrapper" for the application to consume it. Guess i could share that boilerplate code, but not the application that consumes it.

**Comment by cjayyy** [comment]
+1
Due to GPL we are not allowed to use the library in our project (even though we would be happy to pay for it).

**Comment by anentropic** [comment]
AFAICT if I run a bundler to concat this lib with other JS in my app I have effectively GPL'd my app, this is not an option for commercial project

I don't mind that the lib is GPL, my question is whether it is also possible to to obtain a non-GPL licensed version?

(the lib looks great, has every feature I need!)

**Reply by anentropic** [reply]
<img width="509" alt="Screenshot 2024-06-05 at 15 23 15" src="https://github.com/easepick/easepick/assets/147840/8dc2b8d7-956f-4961-b478-38245d3d874d">

https://easepick.com/support/contact.html

it this just a broken form, or there used to be a commercial license option but no longer?

**Reply by anentropic** [reply]
this project is dead I guess 🤷‍♂️
